api-scanner/src/test/java/net/corda/plugins/KotlinExcludeMethodTest.java

```diff
@@ -0,0 +1,27 @@
+package net.corda.plugins;
+
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+import org.junit.rules.TemporaryFolder;
+import org.junit.rules.TestRule;
+
+import java.io.IOException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class KotlinExcludeMethodTest {
+
+    private static final TemporaryFolder testProjectDir = new TemporaryFolder();
+    private static final GradleProject testProject = new GradleProject(testProjectDir, "kotlin-exclude-method");
+
+    @ClassRule
+    public static final TestRule rules = RuleChain.outerRule(testProjectDir).around(testProject);
+
+    @Test
+    public void testFilteredMethodsAreExcluded() throws IOException {
+        assertThat(testProject.getApiText())
+                .doesNotContain("<init>(String, String, kotlin.jvm.internal.DefaultConstructorMarker)");
```

Comment:
This test would also pass if the API text were completely empty, though :cry:. So perhaps also test that
```java
.contains("<init>(kotlin.jvm.internal.DefaultConstructorMarker)");
```
You can chain these together so that `getApiText()` is still only called once.

Reply:
good idea, I'll add the class name.